external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -104,6 +104,10 @@ public KafkaSpout(KafkaSpoutConfig<K, V> kafkaSpoutConfig) {
         this.kafkaSpoutConfig = kafkaSpoutConfig;
     }
 
+    long getNumUncommittedOffsets() {
```

Comment:
I'm wondering if we can avoid inspecting this directly in the test by unit testing OffsetManager instead of indirectly testing it via the KafkaSpout? Testing that the spout will commit past gaps in the tuple stream is a good idea, but the check of this field is actually testing that OffsetManager.commit isn't buggy, which I feel belongs in a test for that class.